packages/dcos-integration-test/extra/test_metrics.py

```diff
@@ -170,6 +170,22 @@ def check_adminrouter_metrics():
         check_adminrouter_metrics()
 
 
+def test_metrics_diagnostics(dcos_api_session):
+    """Assert that Admin Router metrics on master are present."""
+    @retrying.retry(wait_fixed=STD_INTERVAL, stop_max_delay=METRICS_WAITTIME)
+    def check_diagnostics_metrics(host):
+        response = get_metrics_prom(dcos_api_session, host)
+        for family in text_string_to_metric_families(response.text):
+            for sample in family.samples:
+                if sample[0].startswith('bundle_creation_time_seconds'):
+                    assert sample[1]['dcos_component_name'] == 'DC/OS Diagnostics'
+                    return
+        raise Exception('Expected DC/OS Diagnostics metrics not found')
+    check_diagnostics_metrics(dcos_api_session.slaves[0])
```

Comment:
might prefer following the pattern from other tests here for consistency and just in case there are no slaves/public slaves even if it's rare (like the block [here](https://github.com/dcos/dcos/pull/4700/files#diff-a99db445cccf205fe11239e4be6de2aaL53))

Reply:
I extracted this to a function